ksql-cli/src/main/java/io/confluent/ksql/cli/StandaloneExecutor.java

```diff
@@ -17,53 +17,38 @@
 package io.confluent.ksql.cli;
 
 
-import io.confluent.ksql.metastore.MetaStore;
-import io.confluent.ksql.util.KafkaTopicClientImpl;
-import io.confluent.ksql.util.KsqlConfig;
-import io.confluent.ksql.util.Pair;
 import io.confluent.ksql.util.PersistentQueryMetadata;
 import io.confluent.ksql.util.QueryMetadata;
 
-import org.apache.kafka.clients.admin.AdminClient;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.Collections;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
-import java.util.concurrent.ExecutionException;
 
 import io.confluent.ksql.KsqlEngine;
-import io.confluent.ksql.parser.tree.Statement;
 
 public class StandaloneExecutor {
 
   private static final Logger log = LoggerFactory.getLogger(StandaloneExecutor.class);
 
   private final KsqlEngine ksqlEngine;
 
-  public StandaloneExecutor(Map streamProperties) throws ExecutionException, InterruptedException {
-    KsqlConfig ksqlConfig = new KsqlConfig(streamProperties);
-    ksqlEngine = new KsqlEngine(ksqlConfig, new KafkaTopicClientImpl(AdminClient.create(ksqlConfig.getKsqlAdminClientConfigProps())));
+  public StandaloneExecutor(final KsqlEngine ksqlEngine) {
+    this.ksqlEngine = ksqlEngine;
   }
 
   public void executeStatements(String queries) throws Exception {
-    MetaStore tempMetaStore = ksqlEngine.getMetaStore().clone();
-    List<Pair<String, Statement>> queryList = ksqlEngine.parseQueries(queries,
-                                                                      Collections.emptyMap(),
-                                                                      tempMetaStore);
-    List<QueryMetadata> queryMetadataList = ksqlEngine.planQueries(
-        queryList, new HashMap<>(), tempMetaStore);
+    final List<QueryMetadata> queryMetadataList = ksqlEngine.createQueries(queries);
     for (QueryMetadata queryMetadata: queryMetadataList) {
       if (queryMetadata instanceof PersistentQueryMetadata) {
         PersistentQueryMetadata persistentQueryMetadata = (PersistentQueryMetadata) queryMetadata;
-        persistentQueryMetadata.getKafkaStreams().start();
+        persistentQueryMetadata.start();
       } else {
-        System.err.println("Ignoring statemenst: " + queryMetadata.getStatementString());
-        System.err.println("Only CREATE statements can run in KSQL embedded mode.");
-        log.warn("Ignoring statemenst: {}", queryMetadata.getStatementString());
-        log.warn("Only CREATE statements can run in KSQL embedded mode.");
+        final String message = String.format("Ignoring statements: %s" +
+                "\nOnly CREATE statements can run KSQL embedded mode.",
+            queryMetadata.getStatementString());
+        System.err.println(message);
```

Comment:
Do we know why we are printing directly to System.err and logging here? Is it just an accident of history? Is it worth removing one of them? It seems like an odd pattern.

Reply:
I'm not sure, but i assumed it was to print to the console. @hjafarpour any idea?